Software/Development/SemesterProject/App/closedLoopControlManager.c

```diff
@@ -3,38 +3,56 @@
  *
  *  Created on: 25. apr. 2019
  *      Author: Nicolai Fransen
+ *
+ *      This module contains the closed loop control algorithm of the system
+ *
+ *      It collects all the different parts of the control system and calls
+ *      them in the correct order.
+ *
+ *      The flow is:
+ *          - Obtain references
+ *          - Obtain currents
+ *          - Obtain Rotor position angle in radians
+ *          - Transform currents from ABC -> dq
+ *          - Calculate dq voltage references from current PI-controllers
+ *          - Transform voltages from dq -> alpha/beta
+ *          - Run SVM with alpha/beta voltages as references.
  */
 
 #include "closedLoopControlManager.h"
 
 void runClosedLoopControl(void)
 {
     float movementReference = 0;
-    float iqReference = 0, idReference = 0;
+    float currentReferences[1];
     float abcCurrents[2];
     float theta = 0;
-    volatile dqObject dqCurrents, dqVoltages;
+    dqObject dqCurrents, dqVoltages;
     alphaBetaObject abVoltages;
 
-    movementReference = getMovementReference();         // Reads torque or speed reference
+    // Checks if the system is in cruise or torque control, and returns
+    // the reference corresponding to that control type.
+    movementReference = getMovementReference();
 
-    iqReference = getIqReference(movementReference);    // Calculate iq reference from movement reference
-    idReference = calculateIdReference();               // Calculate id reference
+    currentReferences[0] = getIqReference(movementReference);   // Calculate iq reference from movement the reference.

```

Comment:
Why not using dqObject for this 2 guys instead of an array? I think it would be cleaner.

Also the comment has a misplaced 'the'